datetime/mod.ts

```diff
@@ -22,14 +27,22 @@ export function parseDate(dateStr: string, format: string): Date {
   return new Date(Number(y), Number(m) - 1, Number(d));
 }
 
+export type DateTimeFormat = 
+  "mm-dd-yyyy hh:mm"
+  | "dd-mm-yyyy hh:mm" 
+  | "yyyy-mm-dd hh:mm"
+  | "hh:mm mm-dd-yyyy"
+  | "hh:mm dd-mm-yyyy"
+  | "hh:mm yyyy-mm-dd"
+
 /**
  * Parse date & time from string using format string
  *
  * @param {string} dateStr - date & time string
- * @param {string} format - format string
+ * @param {DateTimeFormat} format - format string
  * @return {Date} Parsed date
  */
-export function parseDateTime(datetimeStr: string, format: string): Date {
+export function parseDateTime(datetimeStr: string, format: DateTimeFormat): Date {
   let m, d, y, ho, mi: string;
 
   if (format === "mm-dd-yyyy hh:mm") {
```

Comment:
it seems like there should also be an else block at the end here to throw an exception.

whilst with your change there is no longer a DateTimeFormat that passes through, IMO it would be sensible to throw an error (unknown/unsupported datetimeformat) in an else anyway... even though it's unreachable.
this could come up if someone added a new DateTimeFormat in the future... what do you think?

Reply:
Better to add a runtime check too. We want to support JS as well.

Reply:
@djKooks thanks. Added exception for this~